### ISSUE TITLE
[epic][form][ux] Update each component with consistent style guide 

### ISSUE THREAD

**Issue by fenekku** [inveniosoftware/invenio-app-rdm#459]
Follow observations in https://codimd.web.cern.ch/wLJaq6pFQYaYQgZQaBWbiQ?both#General and in General components sections. Can be done in parallel.

### Additional titles component

- Use the Type button
- mockup
    - the view of adding new additional title
    - show the list of additional titles

### Publication date component
- Have it prefilled with today's date
- This problem might be common for other fields 

### Subjects
- Autocompletion input field
- Have an inline dropdown with the `limit to` options (Guillaume: I am unclear about what is meant here)
- Autocompletion should filter by the `limit to` parameter (Ensure is the case now)
- Guillaume: At Northwestern, we've suddenly had people asking for subjects and keywords split. I don't agree with it, but I don't understand why (we are investigating).
- Guillaume: How should the `limit to` behave when terms from a vocabulary have been chosen and a different vocabulary is now selected for `limit to`: I think the chosen terms should always be shown and only the terms in the dropdown be filtered (but for a while that was not the case). 
    - Zach: I agree with that! The `limit to` should control only the suggestions you get. The chosen items are not affected. The only potential confusion I see is the following:
        - User has `limit to` option to "Filter from A"
        - Searches a term that is not in A, but maybe in B
        - Subject component allows additions if not found
        - User adds their suggestion
        - Now we end up with a custom choice even though the term could be found in the B collection. 
            - Custom terms might need some "smart" cleanup.

### Dates
 
- Use the [Type button](#Type-button)
- mockup
    - the view of adding new date
    - show the list of dates
- do we switch order of type and date?

### Publisher

- label on the left
- P-A: :nail_care: prefill with current institution from settings

### Awards
 
- Vocabulary?
- Similar to Resource type component?
- Need to find notes from discussion about this. Otherwise, 2-step autocomplete

### Related work

- identifiers/scheme field similar to [Identifiers component](#Identifiers-component)?
- Resource type similar to [Resource type](#Resource-Type-component)?
- mockup
    - the view of adding new related field
    - show the list of related fields
- make fit in one line, if can't modal

**Comment by lnielsen** [comment]
For February triage:
- Go through each bullet and figure out if something is still left. For left-overs, create new issues.

**Comment by zzacharo** [comment]
My take on what is still a valid case:

- Publication date:
  - Use as default the "today" date
-  Additional titles component
  - Not sure the "Type" component will improve the UX but if we want that then we need to create a ticket for making mockups.
  - We need to fill in the full row (check screenshots)
  - Find a way not to repeat the "Additional Title" and "Type" labels in every additional title added.
- Subjects
  - Considering we disabled it temporarily, maybe we just need to take the notes we have and dump them in another ticket so we can polish it when we attack it? That ticket would probably require also the subjects vocabulary in the backend plus some new mockups on how to integrate the `limit To` field with the autocomplete dropdown.
- Dates
   - Maybe @fenekku knows more about this but I thought we were happy with the work we did...
   - Maybe the same as additional titles to remove the field labels on every row we add..
-  Publisher
  - Confirm if "prefill with current institution from settings" is still valid
- Awards
  - We need mockups.
  - Disabled for this release so maybe create a ticket with having mockups or at least decide the representation of the component
-  Related work
  - Answer open questions
  - Create mockups
